### ISSUE TITLE
feat: ✨ syn-header: Highlight the header when it is used as a "sticky" header

### ISSUE THREAD

**Issue by schilchSICKAG** [synergy-design-system/synergy-design-system#529]
# Checklist (for the issue creator)

- [x] I have looked at the documentation site and was not able to find the required feature.
- [x] I have looked into the Figma boards and was not able to find the required feature.
- [x] I would be willing to provide a PR for this feature.

# Component

I suggest to add the feature to the following component: syn-header

# Story

As a User using an application built with the Synergy Design System, I would like to have a better visual state representation so that I can see the boundary of the `syn-header` when it is fixed to the top.

This could be archived in multiple ways:

1. By creating a new prop `shadow` and attaching a class on it. Only apply this if no `<syn-side-nav>` is found.
2. ~~By using some JavaScript in the component that uses an intersection observer and checks if the item is the page is currently intersecting and applying a class to the `syn-header`.~~

In both cases, it should show a visual highlight when the header is not on its original position anymore, e.g. because the content is scrolling.

# Acceptance criteria

- [ ] All DoD items successfully completed
- [x] Define the styling for fixed positions in design and dev
- [x] Define how to add the styling (Will be added via property)
- [ ] Apply the styling
- [ ] Create a new story in Storybook and Figma

**Comment by reichst1SICKAG** [comment]
There is a design-shadow-ticket for this already: https://github.com/synergy-design-system/design/issues/79

**Comment by schilchSICKAG** [comment]
Blocked by Design: https://github.com/synergy-design-system/design/issues/79

**Comment by reichst1SICKAG** [comment]
@schilchSICKAG Now done and ready. 
Take a look at: https://www.figma.com/design/bZFqk9urD3NlghGUKrkKCR/branch/nv3f3ZfyRyMg8xrjMsfnFQ/Synergy-Digital-Design-System?node-id=36580-140901&t=ecUApzoA83rEgQlY-11
AND
https://www.figma.com/design/bZFqk9urD3NlghGUKrkKCR/branch/nv3f3ZfyRyMg8xrjMsfnFQ/Synergy-Digital-Design-System?node-id=42864-478896&t=ecUApzoA83rEgQlY-11
